main/src/library/Boxed.flix

```diff
@@ -32,6 +32,8 @@ pub enum Boxed {
     )
 }
 
+instance Immutable[Boxed]
```

Comment:
Any reason not to use `with Immutable`?

Reply:
I guess I was just aiming to be consistent with the surrounding code, but I'm happy to switch to `with` if you prefer. Let me know.

Reply:
I would propose we use maximal derivation, i.e. place it on the enum, when possible.